### ISSUE TITLE
lld Mach-O backend reports a warning when using DWARF 5

### ISSUE THREAD

**Issue by petrhosek** [llvm/llvm-project#51668]
|  |  |
| --- | --- |
| Bugzilla Link | [52326](https://llvm.org/bz52326) |
| Version | unspecified |
| OS | All |
| CC | @dwblaikie,@gkmhub,@int3,@nico,@smeenai,@oontvoo |

## Extended Description 
You can reproduce it with the following:

```
$ cat >a.c <<EOF
int main() {}
EOF
$ clang a.c -gdwarf-5 -fuse-ld=lld
ld64.lld: warning: /var/folders/5t/h2t4fx3d2bg1yfs2pcwgzzlw0056zt/T/a-60c5cf.o: invalid reference to or invalid content in .debug_str_offsets[.dwo]: section offset exceeds section size
```

**Comment by BertalanD** [comment]
Proposed fix: https://reviews.llvm.org/D130559

**Comment by yuanfang-chen** [comment]
I could still reproduce this with upstream Clang instead of Apple Clang.

**Comment by BertalanD** [comment]
I can't reproduce the problem with the commands listed in the original issue. Could you share a repro case with me?

I used Clang and LLD built from commit 0e4378c55e318869ea6404f2351f143e6d17fbfe.

**Comment by yuanfang-chen** [comment]
> I can't reproduce the problem with the commands listed in the original issue. Could you share a repro case with me?
> 
> I used Clang and LLD built from commit [0e4378c](https://github.com/llvm/llvm-project/commit/0e4378c55e318869ea6404f2351f143e6d17fbfe).

I had a stale build. Yeah, it was fixed. Apologies for the false alarm.
